Custom Prune OptionButton

jarolrod opened the pull request:
This introduces a new OptionButton in StorageOptions that represents a custom prune target option. This is shown only when the user has gone into detailed settings and chosen a custom prune target.

Additionally, this moves the logic of setting the default prune value from SettingsStorage into StorageOptions. It is the easy storage option button shown during onboarding that should set this. With the original logic, when onboarding is not shown because the user has already gone through it, the app will set prune to 2 gb again.

| light | dark |
| ----- | ---- |
| <img width="752" alt="Screen Shot 2023-02-01 at 10 30 36 PM" src="https://user-images.githubusercontent.com/23396902/216224800-6bda5263-2661-4eed-b550-7cdf94731408.png"> | <img width="752" alt="Screen Shot 2023-02-01 at 10 30 16 PM" src="https://user-images.githubusercontent.com/23396902/216224815-8cdc3f36-6493-4d80-afc8-c4c26ddb2b5c.png"> |



[![Windows](https://img.shields.io/badge/OS-Windows-green)](https://api.cirrus-ci.com/v1/artifact/github/bitcoin-core/gui-qml/win64/insecure_win_gui.zip?branch=pull/<PR>)
[![Intel macOS](https://img.shields.io/badge/OS-Intel%20macOS-green)](https://api.cirrus-ci.com/v1/artifact/github/bitcoin-core/gui-qml/macos/insecure_mac_gui.zip?branch=pull/<PR>)
[![Apple Silicon macOS](https://img.shields.io/badge/OS-Apple%20Silicon%20macOS-green)](https://api.cirrus-ci.com/v1/artifact/github/bitcoin-core/gui-qml/macos_arm64/insecure_mac_arm64_gui.zip?branch=pull/<PR>)
[![ARM64 Android](https://img.shields.io/badge/OS-Android-green)](https://api.cirrus-ci.com/v1/artifact/github/bitcoin-core/gui-qml/android/insecure_android_apk.zip?branch=pull/<PR>)

jarolrod commented:
Updated from [7c3f94d ](https://github.com/bitcoin-core/gui-qml/commit/7c3f94d8e95ffd815f0d6aa0bd82b079871e4489)to [7b4085d](https://github.com/bitcoin-core/gui-qml/commit/7b4085d3601f8340cbd21e9f61a7bb43e41ebd7c), [Compare](https://github.com/bitcoin-core/gui-qml/compare/7c3f94d8e95ffd815f0d6aa0bd82b079871e4489..7b4085d3601f8340cbd21e9f61a7bb43e41ebd7c)

Changes: rebased over main

johnny9 commented on src/qml/components/StorageSettings.qml:
```diff
@@ -28,7 +38,11 @@ ColumnLayout {
             description: optionsModel.pruneSizeGB
-            onEditingFinished: optionsModel.pruneSizeGB = parseInt(text)
+            onEditingFinished: {
+                root.customStorage = true
```
I think custom storage should only be true if the values don't match the other two options.

johnny9 reviewed (commented):
There is an issue with choosing no prune in the detailed settings and the custom option will be selected and it will show that you want to prune with whatever the prune size was.
| | | 
|--|--|
|![Screenshot from 2023-02-14 00-57-43](https://user-images.githubusercontent.com/985648/218652206-fc016284-6608-4f0f-b59a-aa21e04920e8.png)|![Screenshot from 2023-02-14 00-57-32](https://user-images.githubusercontent.com/985648/218652214-f1cc4a2d-b2d9-4098-a05f-429a30427585.png)|

hebasto commented:
> based on #243

It has been merged. Rebase?

jarolrod commented:
Updated from [7b4085d ](https://github.com/bitcoin-core/gui-qml/commit/7b4085d3601f8340cbd21e9f61a7bb43e41ebd7c)to [35e7eb6](https://github.com/bitcoin-core/gui-qml/commit/35e7eb620f4124c26b89c1ebeb7241f12ebebf90), [compare](https://github.com/bitcoin-core/gui-qml/compare/7b4085d3601f8340cbd21e9f61a7bb43e41ebd7c..35e7eb620f4124c26b89c1ebeb7241f12ebebf90)

changes: rebased over main

jarolrod commented:
@johnny9 will address

jarolrod commented:
updated branch from[ 35e7eb6 ](https://github.com/bitcoin-core/gui-qml/commit/35e7eb620f4124c26b89c1ebeb7241f12ebebf90)to[ c85e334](https://github.com/bitcoin-core/gui-qml/commit/c85e334fa1498030c9e50eb0fac4a1f4404a0c37), [compare](https://github.com/bitcoin-core/gui-qml/compare/35e7eb620f4124c26b89c1ebeb7241f12ebebf90..c85e334fa1498030c9e50eb0fac4a1f4404a0c37)

changes: rebased to resolve conflicts, still a WIP

jarolrod commented:
Updated from from[ c85e334 ](https://github.com/bitcoin-core/gui-qml/commit/c85e334fa1498030c9e50eb0fac4a1f4404a0c37)to[ a8d8b34](https://github.com/bitcoin-core/gui-qml/commit/a8d8b34a4d0cbf20b7459a5f4c63c96e159a1304), [compare](https://github.com/bitcoin-core/gui-qml/compare/c85e334fa1498030c9e50eb0fac4a1f4404a0c37..a8d8b34a4d0cbf20b7459a5f4c63c96e159a1304)

Changes: fixed logic for when a prune OptionButton is checked

hebasto commented:
> Based on https://github.com/bitcoin-core/gui-qml/pull/271 ...

https://github.com/bitcoin-core/gui-qml/pull/271 has just been merged.

jarolrod commented:
Updated from [a8d8b34 ](https://github.com/bitcoin-core/gui-qml/commit/a8d8b34a4d0cbf20b7459a5f4c63c96e159a1304)to [e3856c0](https://github.com/bitcoin-core/gui-qml/commit/e3856c08ab80a4a5a980c57e9f27a519727f4c74), [compare](https://github.com/bitcoin-core/gui-qml/compare/a8d8b34a4d0cbf20b7459a5f4c63c96e159a1304..e3856c08ab80a4a5a980c57e9f27a519727f4c74)

Changes: rebased over main

jarolrod commented:
Updated from [e3856c0 ](https://github.com/bitcoin-core/gui-qml/commit/e3856c08ab80a4a5a980c57e9f27a519727f4c74)to [88a38a1](https://github.com/bitcoin-core/gui-qml/commit/88a38a146e6d422f603784a606e269cc53a918c2), [compare](https://github.com/bitcoin-core/gui-qml/compare/e3856c08ab80a4a5a980c57e9f27a519727f4c74..88a38a146e6d422f603784a606e269cc53a918c2)

changes: rebased over main and rebased over changes to PR this is based on.
